lib/rucio/daemons/rsedecommissioner/profiles/generic.py

```diff
@@ -0,0 +1,489 @@
+    rse: 'dict[str, Any]',
+    *,
+    logger: 'Callable[..., None]' = logging.log,
+) -> None:
+    """Check for remaining replicas at the RSE and resolve where possible.
+
+    :param rse: RSE table entry as a dictionary.
+    :param logger: Logging function.
+    :returns: A boolean flag indicating the completion status.
+    """
+    initial_limit = 100
+    replicas = list(list_replicas_per_rse(rse['id'], limit=initial_limit))
+    num_replicas = len(replicas)
+
+    if len(replicas) == 0:
+        # The RSE is now really completely cleared.
+        logger(logging.INFO,
+               '(%s) Completed deleting the RSE contents. Updating the'
+               ' decommissioning status to "done".', rse['rse'])
+
+        set_status(rse['id'], DecommissioningStatus.DONE)
+
+    else:
+        logger(logging.WARNING,
+               '(%s) %s replicas remain on the RSE even though there are no more'
+               ' rules creating locks on them. Attempting automatic cleaning.',
+               rse['rse'],
+               num_replicas if num_replicas < initial_limit else f'>={initial_limit}')
+
+        if num_replicas >= initial_limit:
+            replicas = list_replicas_per_rse(rse['id'])
+
+        _process_replicas_with_no_locks(rse, replicas, logger=logger)
+
+        # We now wait for reaper to pick the updated replicas (if there are any).
+        # Hopefully all replicas will be deleted at the next cycle
+
+
+def _process_replicas_with_no_locks(
+    rse: 'dict[str, Any]',
+    replicas: 'Iterable[dict[str, Any]]',
+    *,
+    logger: 'Callable[..., None]' = logging.log,
+) -> None:
+    """Process replicas that remain at the RSE after the decommissioning.
+
+    Replicas that remain at the RSE at this point should be unlocked. While
+    this is a situation that should not happen, it can due to various errors.
+    This function checks each such replica and attempt to resolve the invalid
+    state for known cases.
+
+    :param rse: RSE table entry as a dictionary.
+    :param replicas: Sequence or generator of replica objects.
+    :param logger: Logging function.
+    """
+    num_processed = 0
+    num_updated = 0
+    num_waiting_reap = 0
+
+    # Check if each replica satisfies the condition in
+    # core.replica.list_and_mark_unlocked_replicas()
+    # For each of the conditions, in case of failure, handle cases that can be
+    # handled
+    for replica in replicas:
+        num_processed += 1
+
+        updated = False
+
+        # If the replica fails any of the following three conditions, reaper
+        # won't pick it up
+        reap_pending = True
+
+        # 1. Condition on the replica state & update time
+        in_stable_state = replica['state'] in [ReplicaState.AVAILABLE,
+                                               ReplicaState.UNAVAILABLE,
+                                               ReplicaState.BAD]
+
+        ten_minutes_ago = datetime.utcnow() - datetime.timedelta(seconds=600)
```

Comment:
You need to import `timedelta` properly and use it directly.

Reply:
Done.
`from datetime import datetime, timedelta`